mindmeld/components/question_answerer.py

```diff
@@ -227,6 +236,23 @@ def load_kb(self, index_name, data_file, **kwargs):
             raise ValueError(msg)
         self._load_kb(index_name, data_file, **kwargs)
 
+    def dump(self, pkl_path):
+        folder = os.path.dirname(pkl_path)
+        os.makedirs(folder, exist_ok=True)
+        self._dump(pkl_path)
+
+        metadata = {
+            "app_path": self.app_path,
+            "app_namespace": self.app_namespace,
+            "resource_loader": self.resource_loader,
```

Comment:
the app_path can be derived from the resource_loader. Moreover, the resource_loader is a huge object with tons of references and gazetteer information that can be derived from the app files itself, so a lot of duplicated memory is going to be stored repeatedly to disk.

Is there more specific information in the resource loader that we can dump instead of the whole object?

Reply:
Thanks for pointing it out. It is unnecessary to dump resource loader and the can be created during instantiation through `ResourceLoader.create_resource_loader(self.app_path)` (app being obtained from the loaded pkl file)

Reply:
removed resource loader from question answerer dump as well as question answerer itself